### DISCUSSION TITLE
Looking for help understanding the Twitch API

### DISCUSSION THREAD

**Discussion by shmup** [Xithrius/twitch-tui#192] [Q&A] [answered]
I know you're going to eventually have a feature that I'm interested in, ala https://github.com/Xithrius/twitch-tui/issues/147

But in the meantime, I'm trying to implement it along with my script discussed here: https://github.com/Xithrius/twitch-tui/discussions/186

I understand oauth needs to take me Elsewhere to authorize, and then send its response to a Return URI. 

I guess I'm wondering, does mean I must stand up even the most minimal little server (`python -m`) or the like?

Here's what I've got, and `#2` is where I'm stuck:

```bash
#!/usr/bin/env bash

# using httpie and jq
# https://dev.twitch.tv/docs/api/reference#get-followed-streams

#1 - pukes out {"access_token":"blahblahblah","expires_in":5609748,"token_type":"bearer"}
TOKEN=$(https -b POST \
  "https://id.twitch.tv/oauth2/token?client_id=$TWITCH_CLIENT_ID&client_secret=$TWITCH_CLIENT_SECRET&grant_type=client_credentials" \
  'Content-Type: application/x-www-form-urlencoded' | jq '.access_token')

#2 - Is it definitely required on my part to stand up a little web server to receive this request?
#    Do I have _any_ other viable options?
https GET "https://id.twitch.tv/oauth2/authorize?response_type=token&client_id=$CLIENT_ID&redirect_uri=http://localhost:3000&scope=user%3Aread%3Afollows"

#3 - this'll work fine when i tackle #2
https GET "https://api.twitch.tv/helix/streams/followed?user_id=$TWITCH_USER_ID" \
  "Authorization: Bearer $TOKEN" \
  "Client-Id: $TWITCH_CLIENT_ID"
```
Any guidance would be appreciated. 

(If you saw me hop into the Python Discord and ask this, it was only afterwords I realized "this is the wrong way to go about this" and left, to post it here)

**Comment by Xithrius** [comment]
Greetings. I'll look into this soon (am working on [other project](https://github.com/xithrius/xithrius.cloud)), thank you for the post.

**Reply by shmup** [reply]
I will explore this repo and probably get the thing I'm looking for: https://github.com/krathalan/wtwitch (not using it as a solution, but seeing Their solution)

You're working hard as hell on `twt` lately, congrats about that!

**Reply by Xithrius** [reply]
Good luck with figuring that out m8.

Finally was able to get abstraction of windows working after a while.

**Reply by Xithrius** [reply]
Perhaps this can help you? https://github.com/nbgibson/streamers/blob/main/streamers/streamers.py

**Comment by Xithrius** [comment, marked as answer]
Tracking in #147, being implemented in #419.
